src/solvers/sat/satcheck_minisat2.h

```diff
@@ -23,57 +24,60 @@ class Solver; // NOLINT(readability/identifiers)
 class SimpSolver; // NOLINT(readability/identifiers)
 }
 
-template<typename T>
 class satcheck_minisat2_baset:public cnf_solvert
 {
 public:
-  explicit satcheck_minisat2_baset(T *);
+  static const double default_random_seed;
+  static const double default_random_var_freq;
+
   virtual ~satcheck_minisat2_baset();
 
-  virtual resultt prop_solve() override;
-  virtual tvt l_get(literalt a) const final;
+  resultt prop_solve() override final;
+  tvt l_get(literalt a) const override final;
 
-  virtual void lcnf(const bvt &bv) final;
-  virtual void set_assignment(literalt a, bool value) override;
+  void lcnf(const bvt &bv) override final;
+  void set_assignment(literalt a, bool value) override final;
 
   // extra MiniSat feature: solve with assumptions
-  virtual void set_assumptions(const bvt &_assumptions) override;
-
-  // extra MiniSat feature: default branching decision
-  void set_polarity(literalt a, bool value);
-
-  virtual bool is_in_conflict(literalt a) const override;
-  virtual bool has_set_assumptions() const final { return true; }
-  virtual bool has_is_in_conflict() const final { return true; }
+  void set_assumptions(const bvt &_assumptions) override final;
 
-  void set_time_limit_seconds(uint32_t lim) override
-  {
-    time_limit_seconds=lim;
-  }
+  bool is_in_conflict(literalt a) const override;
+  bool has_set_assumptions() const override final { return true; }
+  bool has_is_in_conflict() const override final { return true; }
 
 protected:
-  T *solver;
-  uint32_t time_limit_seconds;
-
+  satcheck_minisat2_baset(
+    std::unique_ptr<Minisat::Solver> solver,
+    double random_seed,
+    double random_freq);
+  std::unique_ptr<Minisat::Solver> solver_;
   void add_variables();
-  bvt assumptions;
+
+private:
+  resultt prop_solve_impl();
+
+  bvt assumptions_;
```

Comment:
Why rename variables?  Our coding standard doesn't require private variables to be named with underscores.